go/ai-proxy/main.go

```
@@ -20,43 +19,34 @@ import (
 func main() {
 	klog.V(log.LogLevelMinimal).InfoS("Starting AI Proxy", "provider", args.Provider(), "version", environment.Version, "commit", environment.Commit)
 
+	router := mux.NewRouter()
 	p, err := proxy.NewOllamaTranslationProxy(args.Provider(), args.ProviderHost(), args.ProviderCredentials())
 	if err != nil {
-		klog.ErrorS(err, "Could not create proxy")
-		os.Exit(1)
-	}
-
-	op, err := proxy.NewOpenAIProxy(api.ProviderOpenAI, args.ProviderHost(), args.ProviderCredentials())
-	if err != nil {
-		klog.ErrorS(err, "Could not create proxy")
-		os.Exit(1)
-	}
-
-	eop, err := proxy.NewOpenAIEmbeddingsProxy(api.ProviderOpenAI, args.ProviderHost(), args.ProviderCredentials())
-	if err != nil {
-		klog.ErrorS(err, "Could not create proxy")
-		os.Exit(1)
+		if args.Provider() == api.ProviderBedrock {
```

`if args.Provider() != api.ProviderBedrock`

---

if it's bedrock i want it NoOp bc there's no Ollama to Bedrock proxy and therefore it'll always fail. But if it's _not_ bedrock and there's an err, it should fail becasue then that's truly unexpected

---

yea and if there was just an if not clause you don't need the goofy else